src/client.rs

```diff
@@ -57,33 +80,34 @@ impl SpanStack {
         self.current_span_stack.push_back(id);
     }
 
-    // Move span to "completed" and return true if it's time to send, or false to keep
-    // buffering the spans.  Unless already set, automatically set the span's parent if there are
-    // still "current" spans after popping the current one (indicating that the next on the
-    // stack is the current span's "parent").
-    fn end_span(&mut self, span: Span) -> bool {
+    // Move span to "completed" and return the current span ID of the back of the stack (or None
```

Comment:
I would prefer it you called it the top of the stack. WDYT? Also "of" should be "off"

Reply:
I don't have a huge preference about "back" (as in "push_back") or "top".  As for "of", I think I meant it as "of".  It's not the ID of the span that got popped off.  It's the ID of the span that's still on the top (or back) of the stack (if present).